lib/screens/v2/widget_instance/alert.ex

```diff
@@ -2,85 +2,210 @@ defmodule Screens.V2.WidgetInstance.Alert do
   @moduledoc false
 
   alias Screens.Alerts.Alert
+  alias Screens.Config.Screen
+  alias Screens.Config.V2.Departures.Query.Params
+  alias Screens.Config.V2.{BusEink, BusShelter, GlEink}
+  alias Screens.RouteType
 
-  alias __MODULE__, as: T
+  defstruct ~w[screen alert stop_sequences active_routes_at_stop]a
 
-  defstruct screen: nil,
-            alert: nil
+  @type stop_id :: String.t()
 
-  @type config :: Screens.V2.ScreenData.config()
+  @type route_id :: String.t()
 
-  @type t :: %T{
-          screen: config(),
-          alert: Alert.t()
+  @type t :: %__MODULE__{
+          screen: Screens.Config.Screen.t(),
+          alert: Alert.t(),
+          stop_sequences: list(list(stop_id())),
+          active_routes_at_stop: list(route_id())
         }
 
-  @base_alert_priority 1
-
-  @effect_priorities cancellation: 0,
-                     delay: 1,
-                     detour: 0,
-                     no_service: 0,
-                     shuttle: 0,
-                     suspension: 0,
-                     station_closure: 0,
-                     stop_closure: 0,
-                     stop_moved: 0
-
-  def priority(%T{} = t, alert_active? \\ &Alert.happening_now?/1) do
-    [
-      @base_alert_priority,
-      active_priority(t, alert_active?),
-      informed_entity_priority(t),
-      effect_priority(t)
-    ]
-  end
-
-  def serialize(%T{} = t, alert_active? \\ &Alert.happening_now?/1) do
-    %{
-      pill: :bus,
-      icon: :warning,
-      active_status: if(alert_active?.(t.alert), do: :active, else: :upcoming),
-      header: serialize_header(t),
-      text: serialize_text(t)
+  def active?(%__MODULE__{alert: alert}, happening_now? \\ &Alert.happening_now?/1) do
+    happening_now?.(alert)
+  end
+
+  @spec seconds_from_onset(t(), DateTime.t()) :: integer() | :infinity
+  def seconds_from_onset(t, now \\ DateTime.utc_now())
+
+  def seconds_from_onset(%__MODULE__{alert: %Alert{active_period: [{start, _} | _]}}, now)
+      when not is_nil(start) do
+    DateTime.diff(now, start, :second)
+  end
+
+  def seconds_from_onset(_t, _now), do: :infinity
+
+  @spec seconds_to_next_active_period(t(), DateTime.t()) :: integer() | :infinity
+  def seconds_to_next_active_period(t, now \\ DateTime.utc_now())
+
+  def seconds_to_next_active_period(
+        %__MODULE__{alert: %Alert{active_period: active_periods}},
+        now
+      )
+      when not is_nil(active_periods) do
+    next_active_period =
+      Enum.find(active_periods, fn {start, _} ->
```

Comment:
Is `active_periods` guaranteed to be sorted?

Reply:
Thanks for catching this! I updated the alert parser to sort active periods up front!